onnxruntime/core/session/inference_session.cc

```diff
@@ -955,12 +1020,45 @@ class InferenceSession::Impl {
     return Status::OK();
   }
 
+  Status ValidateLevel(unsigned int level) const {
+    auto t_level = (unsigned int)std::pow(2, level);
+    if (t_level < TransformerLevel::MaxTransformerLevel) {
+      return Status::OK();
+    }
+    return Status(ONNXRUNTIME, FAIL, "Unsupported level " + level);
+  }
+
+  void SetTransformerContext(uint32_t& transformers_enabled, std::vector<TransformerLevel>& all_levels, const uint32_t& level) {
```

Comment:
>std::vector<TransformerLevel>& all_levels [](start = 61, length = 41)

Given you go to all the trouble to use TransformerLevel as a bitmask, why pass a vector instead of setting multiple values in a single value?